### ISSUE TITLE
Optimizing legal move generation

### ISSUE THREAD

**Issue by alfredbirk** [OyvindSabo/quoridor.js#5]
From what I can see, to generate all legal moves in a position, a brute-force approach of trying all walls and then using A-star is used. Section 4.2 of [this](https://drive.google.com/file/d/18Yb-77b7WjIwPpgJq4m2cqwvYUbcQkvw/view?usp=sharing) excellent thesis contains some ideas to make this even faster. Fast performance of generating legal moves could be useful for those  who for example wants to use the library to make an AI.

**Comment by OyvindSabo** [comment]
Yup, I will prioritize getting everything to work in as simple a way as possible first, and once it's "done" I will start optimizing. This thesis looks excellent indeed. Any idea who wrote it?

**Comment by alfredbirk** [comment]
Good plan 👍  No idea who wrote it, be must be some kind of legend. What a masterpiece!
